sdk/tests/iot/provisioning/test_az_iot_provisioning_client_parser.c

```diff
@@ -27,12 +27,18 @@
 #define TEST_STATUS_FAILED "failed"
 #define TEST_STATUS_DISABLED "disabled"
 #define TEST_STATUS_UNASSIGNED "unassigned"
+#define TEST_STATUS_UNKNOWN "unknown status123!@#"
 
 #define TEST_ERROR_MESSAGE_INVALID_CERT "Invalid certificate."
 #define TEST_ERROR_MESSAGE_ALLOCATION "Custom allocation failed with status code: 400"
 #define TEST_ERROR_TRACKING_ID "8ad0463c-6427-4479-9dfa-3e8bb7003e9b"
 #define TEST_ERROR_TIMESTAMP "2020-04-10T05:24:22.4718526Z"
 
+// internal APIs
+AZ_NODISCARD az_result _iot_provisioning_client_parse_operation_status(
+    az_span response_operation_status,
+    az_iot_provisioning_client_operation_status* out_operation_status);
+
```

Comment:
If you really want to test this, consider moving the actual `_iot_provisioning_client_parse_operation_status` to an internal header and include that header in tests. This way, there is no need to remember updating in two places.

Otherwise, I suggest just remove the usage from the  tests and only indirectly test this (now implementation detail) via the public API.